core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -234,9 +236,14 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val actorSyste
         properties += ((key, value))
       }
     }
+
     // TODO (prashant) send conf instead of properties
     driverActor = actorSystem.actorOf(
       Props(new DriverActor(properties)), name = CoarseGrainedSchedulerBackend.ACTOR_NAME)
+
+    // If a principal and keytab have been set, use that to create new credentials for executors
+    // periodically
+    SparkHadoopUtil.get.scheduleLoginFromKeytab()
```

Comment:
Ah, I see why you exposed that method that way. Probably ok, but it does feel a little weird; I'd expect the caller to know that it needs to schedule this thing, and if not running on Yarn, things should blow up.

(But I guess they already blow up because the command line options prevent that?)